content/docs/user-guide/programming/memory-management.md

```diff
@@ -10,72 +10,17 @@ When managing memory in O3DE, use AZ memory management calls and avoid static va
-
-Remember that the lifetime of a static variable lasts until the module in which the variable is declared is unloaded. In static linked libraries, the duration of static variables is the entire duration of the application.
-
-### Static Local Variable Example
-
-The following example code from AzFramework \(`\dev\Code\Framework\AzFramework\AzFramework\Physics\DefaultDebugDrawSettings.h`\) uses function static container variables whose destructors rely on `AZ::SystemAllocator`.
-
-**Example**
-
-```
-namespace Physics
-{
-    AZ_INLINE void GetDefaultDebugDrawSettings(DebugDrawSettings& settings)
-    {
-        static AZStd::vector<Vec3> verts;      // These function static container variables use AZ::SystemAllocator.
-        static AZStd::vector<ColorB> colors;
-        //...
-    }
-}
-```
-
-By default, the `AZStd::vector` class uses the `SystemAllocator`. Invoking this function on application shutdown, when `SystemAllocator` no longer exists, causes a deletion from a destroyed heap.
-
-### Global Static Variable Example 
-
-On startup, the following code attempts to create a global static variable whose constructor connects to an EBus:
-
-**Example**
-
-```
-// Example of a global static variable that connects to an EBus.
-namespace LmbrCentral
-{
-    // Wraps an IMaterial pointer so that BehaviorContext can use it.
-    class MaterialHandle
-        : public AZ::RenderNotificationsBus::Handler
-    {
-    public:
-        AZ_CLASS_ALLOCATOR(MaterialHandle, AZ::SystemAllocator, 0);
-        AZ_TYPE_INFO(MaterialHandle, "{BF659DC6-ACDD-4062-A52E-4EC053286F4F}");
-        MaterialHandle()
-        {
-            AZ::RenderNotificationsBus::Handler::BusConnect();
-        }
-//...
-// Later in the code, a MaterialHandle in global space is declared:
-MaterialHandle g_defaultMaterialHandle;
-```
-
-The code crashes the engine as soon as the module that contains the global variable definition is loaded. The constructor of the global variable attempts to connect to the EBus, but the EBus system is not ready because it uses an environment variable from a module that has not yet been initialized.
+O3DE initializes its memory system via storing an instances of Allocators in an AZ Environment variables the first time a call to `AZ::AllocatorInstance<AllocatorType>::Get()` is called in [code](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Memory/AllocatorInstance.h#L73-L86).  
+To ensure that Allocators are available in static initialization and static de-initialization of the running applicaition and any dynamically loaded Gem modules, the allocator AZ Environment variables are stored in the `O3DEKernel` shared library which link library dependency of the [AzCore Target](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/CMakeLists.txt#L60-L98).  
+As the `AzCore` static library target is a dependency by Gems and O3DE applications, they all statically link the `O3DEKernel` target due to that dependency.  
```

Comment:
Possible rewording?
```suggestion
All Gems and O3DE applications statically link the `O3DEKernal` target because they  depend on the `AzCore` static library target.
```

Reply:
The re-write sounds good.